packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -224,14 +225,48 @@ export class FieldSchema<
 	protected _typeCheck?: MakeNominal;
 
 	/**
-	 * @param kind - The [kind](https://en.wikipedia.org/wiki/Kind_(type_theory)) of this field.
-	 * Determine the multiplicity, viewing and editing APIs as well as the merge resolution policy.
-	 * @param allowedTypes - What types of tree nodes are allowed in this field.
+	 * Lazily evaluated allowed types.
 	 */
+	private readonly lazyTypes: Lazy<AllowedTypes>;
+
+	/**
+	 * What types of tree nodes are allowed in this field.
+	 *
+	 * @remarks
+	 * This is the same set of types in {@link FlexFieldSchema.allowedTypes}, just as a set with
+	 * laziness removed.
+	 */
+	public get allowedTypeSet(): AllowedTypes {
```

Comment:
This isn't a set, and its type still has lazyness. I think Set<TreeNodeSchema> is what we want here.

Note that if you don't do that (and keep returning `LazyItem<TreeNodeSchema>`, then there is no reason for lazyTypes  to be lazy. You only need that to be lazy since toy can't eagerly evaluate the LazyItems.

Reply:
And yeah, I meant to ask about the laziness pattern. Wasn't sure if there was something implicit going on that needed the pattern. Removed now.

Reply:
Is there a use case that would prefer AllowedTypes over having a `ReadonlySet<TreeNodeScehma>` (Implemented using the lazy setup)? That seems significantly more useful as you can efficiently check if a type is allowed in the field, and iterating over the allowed schema is easier since you don't have to deal with LazyItem stuff.

Reply:
Fair, probably not. Let me poke at that a bit and see if it cleans things up.

Reply:
Yeah, that cleaned up nicely. Refactored.